src/hayhooks/server/utils/create_valid_type.py

```diff
@@ -36,6 +36,12 @@ def _handle_generics(t_) -> GenericAlias:
             else:
                 new_type[arg_name] = arg_type
         if new_type:
+            # because TypedDict can't handle union types with None
+            # rewrite them as Optional[type]
+            for arg_name, arg_type in new_type.items():
+                type_args = get_args(arg_type)
+                if len(type_args) == 2 and type_args[1] is type(None):
+                    new_type[arg_name] = Optional[type_args[0]]
```

Comment:
Wouldn't this be better in `_handle_generics`? 🤔 

If you have nested `TypedDict`s this would fail in any case because we're checking only the first level right?

Reply:
I could try. Just the code was a little hard to understand and I've decided to do it the end.

Reply:
Ok, if you can't manage I can give it a try. Don't want to spoil all the fun. :)